frontend/app/components/auth/components/oauth-button.tsx

```diff
@@ -0,0 +1,79 @@
+import { h, FunctionComponent, JSX } from 'preact';
+import { useEffect, useState } from 'preact/hooks';
+import { useIntl } from 'react-intl';
+import classnames from 'classnames';
+
+import type { OAuthProvider } from 'common/types';
+import { siteId } from 'common/settings';
+import capitalizeFirstLetter from 'utils/capitalize-first-letter';
+import messages from 'components/auth/auth.messsages';
+
+import styles from './oauth-button.module.css';
+
+const OAUTH_ICONS = {
+  facebook: require('./assets/facebook.svg').default as string,
```

Comment:
Why do you use require instead of import?

Reply:
In my opinion, it looks better in comparison with imports. I don't like the fence of the imports.
The problem with it is we should get `.default`. It could be solved with disabling esmodule for `.svg`. I haven't done it from the first attempt and have left as it is for a better time.

Both approaches are acceptable. It's all only about style in this case. For JS chunks it works in a slightly different way.

```js
const OAUTH_ICONS = {
  facebook: require('./assets/facebook.svg').default as string,
}
```

```js
import facebookIcon from './assets/facebook.svg'

const OAUTH_ICONS = {
  facebook: facebookIcon
}
```